admin/controllers/questions.js

```diff
@@ -0,0 +1,48 @@
+const CheckForm = require('../models/check-form')
+const Pupil = require('../models/pupil')
+const School = require('../models/school')
+const { QUESTION_TIME_LIMIT, TIME_BETWEEN_QUESTIONS } = require('../config')
+
+/**
+ * Returns the set of questions, pupil details and school details in json format
+ * @param req
+ * @param res
+ * @returns { object }
+ */
+
+const getQuestions = async (req, res) => {
+  const { pupilId, schoolId } = req.body
+  if (!pupilId || !schoolId) res.status(400).send('invalid input')
+  let checkForm, pupil, school
+  try {
+    // Until we determine the logic behind fetching the appropriate check form
+    // the pupil will receive the first one
+    checkForm = await CheckForm.findOne({}).sort({createdAt: 1}).lean().exec()
+    pupil = await Pupil.findOne({ '_id': pupilId }).lean().exec()
+    school = await School.findOne({_id: schoolId}).lean().exec()
+  } catch (error) {
+    throw new Error(error)
+  }
+  if (!checkForm) res.status(404).send('Question set not found for pupil')
+  if (!pupil || !school) res.status(401).send('Unauthorised')
```

Comment:
@jon-shipley the responses are now json documents. The only way to access the questions would be if they know the pupil pin and school pin and pass them as request payload. Otherwise 401 unauthorized is thrown